atlp-rwanda/ecommerce-app-predators-bn
# Implement update-product-from-collection feature (#33)

What does this PR do?

This pull request implements the update-product-from-collection feature in the backend of the application.
Description of Task to be completed

Implement an API endpoint that updates a product in the collection based on user input.
Add error handling to ensure that only valid data can be entered.

How should this be manually tested?

Start the backend server by running the following command:
    npm run dev

Send a PUT request to the update-product endpoint with valid input data:

  http://localhost:3000/api/products/:id

Verify that the response status code is 200 and the updated product information is returned.

   {
    "status": "success",
    "message": "Product updated successfully",
    "data": {
        "id": 1,
        "name": "Samsung Galaxy S21",
        "description": "Samsung Galaxy S21 is a smartphone that was tested with the Android 11 operating system. This model weighs 6.2 ounces, has a 6.2-inch touch screen display, 12-megapixel main camera, and 10-megapixel selfie camera. It comes with 8GB of RAM.",
        "price": 1000,
        "quantity": 10,
        "image": "https://images-na.ssl-images-amazon.com/images/I/71ZtNda%2BguL._AC_SL1500_.jpg",
        "createdAt": "2021-03-01T14:00:00.000Z",
        "updatedAt": "2021-03-01T14:00:00.000Z"
    }
}

Send a PUT request with invalid input data and verify that the appropriate error response is returned.
    {
        "status": "error",
        "message": "Validation error",
        "errors": [
            {
                "field": "name",
                "message": "Name is required"
            }
        ]
    }

Any background context you want to provide?

This feature was requested by the product owner to improve the product management capabilities of the backend API.
What are the relevant pivotal tracker stories?

[#33] Implement 'update-product-from-collection' feature in backend API.
Screenshots (if appropriate)

N/A
